lnwallet/channel_test.go

```diff
@@ -4048,6 +4048,168 @@ func TestChannelRetransmissionFeeUpdate(t *testing.T) {
 	}
 }
 
+// TestFeeUpdateOldDiskFormat tests that we properly recover FeeUpdates written
+// to disk using the old format, where the logIndex was not written.
+func TestFeeUpdateOldDiskFormat(t *testing.T) {
+	t.Parallel()
+
+	// Create a test channel which will be used for the duration of this
+	// unittest. The channel will be funded evenly with Alice having 5 BTC,
+	// and Bob having 5 BTC.
+	aliceChannel, bobChannel, cleanUp, err := CreateTestChannels()
+	if err != nil {
+		t.Fatalf("unable to create test channels: %v", err)
+	}
+	defer cleanUp()
+
+	// First, we'll fetch the current fee rate present within the
+	// commitment transactions.
+	startingFeeRate := SatPerKWeight(
+		aliceChannel.channelState.LocalCommitment.FeePerKw,
+	)
+
+	// Next, we'll start a commitment update, with Alice sending a new
+	// update to double the fee rate of the commitment.
+	newFeeRate := startingFeeRate * 2
+
+	// We will send a few HTLCs and a fee update.
+	htlcAmt := lnwire.NewMSatFromSatoshis(0.1 * btcutil.SatoshiPerBitcoin)
+	const numHTLCs = 10
+	var htlcs []*lnwire.UpdateAddHTLC
+	for i := 0; i < numHTLCs; i++ {
+		htlc, _ := createHTLC(i, htlcAmt)
+		if _, err := aliceChannel.AddHTLC(htlc, nil); err != nil {
+			t.Fatalf("unable to add htlc: %v", err)
+		}
+		if _, err := bobChannel.ReceiveHTLC(htlc); err != nil {
+			t.Fatalf("unable to recv htlc: %v", err)
+		}
+		htlcs = append(htlcs, htlc)
+
+		if i != numHTLCs/2 {
+			continue
+		}
+
+		// When half of the HTLCs are sent, also include a fee update.
```

Comment:
Any reason why this just isn't split up to be outside the loop? As is, there'll be multiple fee updates FWIW.